engine/lib/membrane_rtc_engine/engine.ex

```diff
@@ -611,6 +591,22 @@ defmodule Membrane.RTC.Engine do
   #   removed tracks. Also forwards custom media events.
   #
 
+  defp handle_endpoint_notification(:finished, endpoint_id, ctx, state) do
+    case handle_remove_endpoint(endpoint_id, ctx, state) do
+      {{:present, endpoint}, actions, new_state} ->
+        dispatch(%Message.EndpointRemoved{endpoint_id: endpoint_id, endpoint_type: endpoint.type})
+
+        Membrane.Logger.info("Endpoint #{endpoint_id} removed after processing finished")
+
+        {actions, new_state}
+
+      {:absent, actions, new_state} ->
+        Membrane.Logger.info("Endpoint #{endpoint_id} was already removed")
+
+        {actions, new_state}
+    end
+  end
```

Comment:
I think that most of the RTC Engine logs should be on debug as this is a library.  I would use warnings when we notice a behaviour that seems like a RC or may reslut in a bug

```suggestion
  defp handle_endpoint_notification(:finished, endpoint_id, ctx, state) do
    Membrane.Logger.debug("Endpoint: #{endpoint_id} marked itself for removal. Trying to remove.")
    case handle_remove_endpoint(endpoint_id, ctx, state) do
      {{:present, endpoint}, actions, new_state} ->
        dispatch(%Message.EndpointRemoved{endpoint_id: endpoint_id, endpoint_type: endpoint.type})

        Membrane.Logger.debug("Endpoint #{endpoint_id} successfully removed.")

        {actions, new_state}

      {:absent, actions, new_state} ->
        Membrane.Logger.warning("Endpoint #{endpoint_id} marked itself for removal but it has already been removed.")

        {actions, new_state}
    end
  end
```